cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -24,68 +24,149 @@ import (
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// MasterConfiguration contains all necessary information for kubeadm to bootstrap a cluster
 type MasterConfiguration struct {
```

Comment:
Is there a roundtrip_test.go? If not, it should be added. It's just to easy to break stuff.